scripts/release.js

```diff
@@ -0,0 +1,162 @@
+#!/bin/env node
+const { spawnSync } = require("child_process"),
+    readline = require("readline"),
+    fs = require("fs")
+;
+
+const
+  gitRawCommits = require("git-raw-commits"),
+  semver = require("semver"),
+  getStream = require("get-stream"),
+  _ = require("colors")
+;
+
+const _typeReg = /(?<type>[a-zA-Z]+)/;
+const _scopeReg = /(\((?<scope>.*)\))?/;
+const _messageReg = /(?<message>[^(]*)/;
+const _mrReg = /(\(#(?<mr>[0-9]+)\))?/;
+const COMMIT_REGEX = new RegExp(`${_typeReg.source}${_scopeReg.source}: *${_messageReg.source} *${_mrReg.source}`);
+const CHANGELOG_PATH = "./CHANGELOG.md";
+const GO_VERSION_PATH = "./scw/version.go";
+const TMP_BRANCH = "new-release";
+
+function prompt(prompt) {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+  });
+  return new Promise(resolve => {
+    rl.question(prompt, answer => {
+      resolve(answer);
+      rl.close();
+    });
+  });
+}
+
+function execSync(cmd, args) {
+  console.log(`    ${cmd} ${args.join(" ")}`.grey);
+  const { stdout, status, stderr } = spawnSync(cmd, args, { encoding: "utf8" });
+  if (status !== 0) {
+    throw new Error(`return status ${status}\n${stderr}\n`);
+  }
+  return stdout;
+}
+
+function replaceInFile(path, oldStr, newStr) {
+  console.log(`    Editing ${path}`.grey);
+  const content = fs.readFileSync(path, { encoding: "utf8" }).replace(oldStr, newStr);
+  fs.writeFileSync(path, content);
+}
+
+async function main() {
+
+  //
+  // Initialization
+  //
+
+  console.log("Trying to guess gihub username from git remote origin url (only work with ssh)".blue);
+  const githubUsername = execSync("git", ["remote", "get-url", "origin"]).match(/.*:([a-z0-9-]+)\/.*/)[1];
```

Comment:
plot twist: my origin is not my gh username + I rarely name a remote with a username. Let's add catch an env variable?

Reply:
it parse string like `git@github.com:scaleway/scaleway-sdk-go.git`

Reply:
For most people yes, but in my case `git remote get-url origin` is `git@github.com:kindermoumoute/scaleway-sdk-go.git`

Reply:
that ok regex in this case will return `kindermoumoute` this is the expected value